pkg/apis/jaegertracing/v1/options.go

```diff
@@ -38,7 +39,11 @@ func (o *Options) Filter(prefix string) Options {
 // UnmarshalJSON implements an alternative parser for this field
 func (o *Options) UnmarshalJSON(b []byte) error {
 	var entries map[string]interface{}
-	json.Unmarshal(b, &entries)
+	d := json.NewDecoder(bytes.NewReader(b))
+	d.UseNumber()
```

Comment:
A bit of context JSON spec defines numbers as `float64` so all number are by default unmarshalled to `float64`. Our `Options.Args()` methods uses `fmt.Sprintf(%d, val)` to convert the value to string - the `%v` formats the float into exponent notation e.g. `5e+06`

Reply:
Thanks for the details - would it be possible to provide a test?

Reply:
Including one that fails the decode, just to improve the code coverage.

Reply:
Apart from it is it ok?

Reply:
Yes